compact.go

```diff
@@ -59,7 +59,9 @@ type Compactor interface {
 
 	// Compact runs compaction against the provided directories. Must
 	// only be called concurrently with results of Plan().
-	Compact(dest string, dirs ...string) (ulid.ULID, error)
+	// Can optionally pass a list of already open blocks,
+	// to avoid having to reopen them.
+	Compact(dest string, dirs []string, open []*Block) (ulid.ULID, error)
```

Comment:
Since this is changing public interface. Could you tag a minor version? We don't have any tags yet so maybe a good time to start now.

Reply:
0.1.0?

Reply:
Yeah , or maybe even 1.1.0 , since it is pretty stable now and I think go modules don't pull anything below 1.0. It would be nice if we could use these tags as well  when we vendor it in Prometheus.

Reply:
> I think go modules don't pull anything below 1.0

Go modules do work with v0.x.y tags. Cutting a v1 is a pretty heavy commitment and I tend to think that a v0.something is appropriate here.

Reply:
Okay, so I should merge and then add a v0.1.0 tag?

Reply:
yep lets try something as simple as that, unless @simonpasquier has any other recommendations.

I will keep reading how are we suppose to do this properly, but for now it is ok to start with just this.

Reply:
We've changed the API many times before, we have to start somewhere.